src/tests/ftest/harness/launch_setup.py

```diff
@@ -0,0 +1,46 @@
+"""
+(C) Copyright 2022 Intel Corporation.
+
+SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from apricot import TestWithoutServers
+
+from run_utils import run_remote, command_as_user
+from user_utils import get_group_id, get_user_groups
+
+
+class HarnessLaunchSetupTest(TestWithoutServers):
+    """Harness launch.py setup test cases.
+
+    :avocado: recursive
+    """
+
+    def test_harness_launch_setup_users(self):
+        """Verify all expected users are setup correctly by launch.py -u.
+
+        :avocado: tags=all
+        :avocado: tags=vm
+        :avocado: tags=harness
+        :avocado: tags=HarnessLaunchSetupTest,test_harness_launch_setup_users
+        """
+        hostlist_clients = self.get_hosts_from_yaml(
+            "test_clients", "server_partition", "server_reservation", "/run/hosts/*")
+        expected_users = self.params.get('client_users', '/run/*')
+        for user_group in expected_users:
+            user, group = user_group.split(':')
+            self.log.info('Checking if group %s exists', group)
+            gids = get_group_id(self.log, hostlist_clients, group).keys()
```

Comment:
I can't work this out, get_group_id gives a dict of host/gid but this then takes keys() so it will return a tuple of hosts, which you then call list() on below?

Reply:
`get_group_id` returns a dict like `{<gid>: <hosts>}`, so `list(keys())` effectively gives all the unique gids. We don't need the hosts here, but the function returns the dictionary format to be more complete/extensible

Reply:
Ah, so it's `gid: host` not `host: gid`.  That makes a lot more sense.

Reply:
Right. The main reason being that `hosts` is a `NodeSet`, which isn't hashable